src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21814.cs

```diff
@@ -0,0 +1,55 @@
+﻿using NUnit.Framework;
+using NUnit.Framework.Legacy;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues
+{
+	public class Issue21814 : _IssuesUITest
+	{
+		public override string Issue => "Add better parameters for navigation args";
+
+		public Issue21814(TestDevice device) : base(device)
+		{
+		}
+
+		[Test]
+		[Category(UITestCategories.Navigation)]
+		public void VerifyNavigationEventArgs()
```

Comment:
The test running:
![navigation-args-parameters](https://github.com/user-attachments/assets/df14ef41-35f4-4645-958e-30c9a69325ef)